### PR TITLE
Move mhlo.slice op lowering to tensors world.

### PR DESCRIPTION
**Pull request by hanhanW**
- Lower mhlo.slice op to subtensor op in HLOToLinalgOnTensors.cpp
- Lower subtensor op to subview op in HLOToLinalgOnBuffers.cpp

### CODE REVIEW

File: iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp [hunk 1]
```diff
@@ -148,6 +148,43 @@ struct TorchIndexSelectOpConversion
...
+      int64_t stride = op.strides().getValue<int64_t>(i);
+      staticOffsets.push_back(offset);
+      staticSizes.push_back((limit - offset) / stride);
+      staticStrides.push_back(stride);
+    }
+    auto subTensorOp = rewriter.create<SubTensorOp>(
+        loc, args[0], staticOffsets, staticSizes, staticStrides,
+        /*offsets=*/ValueRange{},
+        /*sizes=*/ValueRange{}, /*strides=*/ValueRange{});
+    rewriter.replaceOp(op, subTensorOp.getResult());
```
Comment (antiagainst) [inline iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp:182]: `replaceOpWithNewOp`?

**Review by antiagainst — CHANGES_REQUESTED** [review]
Can we add tests for this change? We are quite under unit tested in general.

File: iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp (hunk 1 above)
Comment (hanhanW) [reply iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp]: done.

**Comment by hanhanW** [comment]
> Can we add tests for this change? We are quite under unit tested in general.

We had a slice.test for testing the pipeline. I split it into slice.mlir and subtensor.mlir. One is testing hlo-to-linalg-on-tensors, and another is testing hlo-to-linalg-on-buffers.

**Review by antiagainst — APPROVED** [review]

**Pull request merged**